pyrit/prompt_converter/confusable_converter.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import logging
+import random
+import re
+from typing import Literal
+
+from confusables import confusable_characters
+from confusable_homoglyphs.confusables import is_confusable
+
+from pyrit.models import PromptDataType
+from pyrit.prompt_converter import PromptConverter, ConverterResult
+
+
+logger = logging.getLogger(__name__)
+
+
+class ConfusableConverter(PromptConverter):
+    """
+    A PromptConverter that applies substitutions to words in the prompt
+    to test adversarial textual robustness by replacing characters with visually similar ones.
+    """
+
+    def __init__(
+        self,
+        *,
+        source_package: Literal["confusable_homoglyphs", "confusables"] = "confusable_homoglyphs",
+        deterministic: bool = False,
+    ):
+        """
+        Initializes the ConfusableGenerator.
+
+        Args:
+            source_package: The package to use for homoglyph generation. Can be either "confusable_homoglyphs"
+                which can be found here: https://pypi.org/project/confusable-homoglyphs/ or "confusables" which can be
+                found here: https://pypi.org/project/confusables/. "Confusable_homoglyphs" is used by default as it is
+                more regularly maintained and up to date with the latest Unicode-provided confusables found here:
+                https://www.unicode.org/Public/security/latest/confusables.txt. However, "confusables"
+                provides additional methods of matching characters (not just Unicode list), so each character
+                has more possible substitutions.
+            deterministic: This argument is for unittesting only.
+        """
+        super().__init__()
+        if source_package not in ["confusable_homoglyphs", "confusables"]:
+            raise ValueError("Invalid source package. Please choose either 'confusable_homoglyphs' or 'confusables'")
```

Comment:
Nit: Could we add `source_package` in the error message for clarity.

```suggestion
            raise ValueError(f"Invalid source package: {source_package}. Please choose either 'confusable_homoglyphs' or 'confusables'")
```